<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <widget class="CodeEdit" name="CodeWindow">
      <property name="font">
       <font>
        <family>Monaco</family>
        <pointsize>18</pointsize>
        <weight>50</weight>
        <bold>false</bold>
       </font>
      </property>
     </widget>
    </item>
    <item row="1" column="0">
     <widget class="console" name="Console">
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>150</height>
       </size>
      </property>
      <property name="font">
       <font>
        <family>Monaco</family>
        <pointsize>16</pointsize>
       </font>
      </property>
     </widget>
    </item>
    <item row="0" column="1" rowspan="2">
     <widget class="DebugWin" name="DebugWindow">
      <property name="maximumSize">
       <size>
        <width>200</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="font">
       <font>
        <family>Monaco</family>
       </font>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="new_file"/>
    <addaction name="open_file"/>
    <addaction name="separator"/>
    <addaction name="save_file"/>
    <addaction name="separator"/>
    <addaction name="actionClose"/>
   </widget>
   <widget class="QMenu" name="menu_2">
    <property name="title">
     <string>Edit</string>
    </property>
    <addaction name="actionundo"/>
    <addaction name="actionRedo"/>
    <addaction name="separator"/>
    <addaction name="actionCopy"/>
    <addaction name="actionPaste"/>
    <addaction name="actionCut"/>
   </widget>
   <widget class="QMenu" name="menuGenerate">
    <property name="title">
     <string>Generate</string>
    </property>
    <addaction name="generate"/>
    <addaction name="generate_coe"/>
    <addaction name="generate_coe_asm"/>
    <addaction name="separator"/>
    <addaction name="assemble"/>
    <addaction name="separator"/>
    <addaction name="disassemble"/>
   </widget>
   <widget class="QMenu" name="menuDebug">
    <property name="title">
     <string>Debug</string>
    </property>
    <addaction name="actionStart"/>
    <addaction name="separator"/>
    <addaction name="actionSingle_step"/>
    <addaction name="actionRun"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionHelp"/>
    <addaction name="actionManual"/>
   </widget>
   <addaction name="menu"/>
   <addaction name="menu_2"/>
   <addaction name="menuGenerate"/>
   <addaction name="menuDebug"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="new_file">
   <property name="text">
    <string>New</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="actionCopy">
   <property name="text">
    <string>Copy</string>
   </property>
  </action>
  <action name="actionPaste">
   <property name="text">
    <string>Paste</string>
   </property>
  </action>
  <action name="open_file">
   <property name="text">
    <string>Open</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="generate_coe">
   <property name="text">
    <string>Generate coe from binary code</string>
   </property>
   <property name="iconText">
    <string>Generate coe from binary code</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+C</string>
   </property>
  </action>
  <action name="save_file">
   <property name="text">
    <string>Save</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionClose">
   <property name="text">
    <string>Close</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+W</string>
   </property>
  </action>
  <action name="actionundo">
   <property name="text">
    <string>Undo</string>
   </property>
  </action>
  <action name="actionRedo">
   <property name="text">
    <string>Redo</string>
   </property>
  </action>
  <action name="actionCut">
   <property name="text">
    <string>Cut</string>
   </property>
  </action>
  <action name="assemble">
   <property name="text">
    <string>Assemble</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+A</string>
   </property>
  </action>
  <action name="disassemble">
   <property name="text">
    <string>Disassemble</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+D</string>
   </property>
  </action>
  <action name="generate_coe_asm">
   <property name="text">
    <string>Generate coe from asm</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+G</string>
   </property>
  </action>
  <action name="generate">
   <property name="text">
    <string>Generate binary code</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+G</string>
   </property>
  </action>
  <action name="actionSingle_step">
   <property name="text">
    <string>Single step</string>
   </property>
   <property name="shortcut">
    <string>F8</string>
   </property>
   <property name="visible">
    <bool>true</bool>
   </property>
  </action>
  <action name="actionStart">
   <property name="text">
    <string>Start</string>
   </property>
   <property name="shortcut">
    <string>F5</string>
   </property>
  </action>
  <action name="actionRun">
   <property name="text">
    <string>Run</string>
   </property>
   <property name="shortcut">
    <string>F9</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>About</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+H</string>
   </property>
  </action>
  <action name="actionManual">
   <property name="text">
    <string>Manual</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+M</string>
   </property>
  </action>
  <action name="actionHelp">
   <property name="text">
    <string>Help</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>CodeEdit</class>
   <extends>QPlainTextEdit</extends>
   <header>CodeEdit.h</header>
  </customwidget>
  <customwidget>
   <class>DebugWin</class>
   <extends>QTableView</extends>
   <header>debugwin.h</header>
  </customwidget>
  <customwidget>
   <class>console</class>
   <extends>QTextBrowser</extends>
   <header location="global">console.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
